src/Codegen/Builders/InputObjectBuilder.hack

```diff
@@ -44,21 +40,47 @@ class InputObjectBuilder extends InputTypeBuilder<\Slack\GraphQL\InputObjectType
             $name_literal = \var_export($field_name, true);
             $type = input_type(($is_optional ? '?' : '').type_structure_to_type_alias($field_ts));
 
+            $fields[$field_name] = shape(
```

Comment:
I was trying to keep this as a constant `FIELDS` but I think for this to work with list inputs, we'll want to do something similar to `getFieldDefinition` so we can take full advantage of the resolved `input_type` type here.

I think this would look something like:

```
function getInputFieldDefinition(string $name): GraphQL\Introspection\InputObjectType::TField {
  switch ($name) {
    case 'name':
      return shape('type' => Types\StringType::nonNullable());
    ...etc
   }
}
```

then in `__InputValue` we can just do:

```
public function getType(): __Type {
   return $this->field['type'];
}
```

and we don't have to leak `is_nullable` etc.

Reply:
Agreed, anything that has a type inside it probably can't be a constant.